web_monitoring/tests/test_html_diff_validity.py

```diff
@@ -94,19 +95,21 @@ def test_html_diff_render_should_not_break_with_empty_content():
 
 def test_html_diff_render_should_raise_for_non_html_content():
     pdf_file = resource_filename('web_monitoring', 'example_data/empty.pdf')
-    pdf_content = Path(pdf_file).read_text(errors='ignore')
-
+    pdf_raw = Path(pdf_file).read_bytes()
+    a = Mock(body=pdf_raw, headers={})
+    b = Mock(body=b'<p>Just a little HTML</p>')
     with pytest.raises(UndiffableContentError):
-        html_diff_render(
-            '<p>Just a little HTML</p>',
-            pdf_content)
+        html_diff_render(_decode_body(a, name='a'),
+                         _decode_body(b, name='b'))
```

Comment:
See my earlier points above about still needing checks in `html_diff_render`.

The way things are written now, the `html_diff_render` call won’t throw an error (or rather, it could throw an error, but it would never be the one we’re testing for). The name of this test is a misnomer and `html_diff_render()` shouldn’t even be involved. If we split up the checks (doing binary decoding checks at the server level, and HTML checks at the method level), we also need to split up the tests—you need unit tests for binary data sent to `_decode_body` and separate unit tests for non-HTML strings/headers sent to `html_diff_render`.

Same comments for all the remaining tests in this PR.